plugins/inventory/vpc_inventory.py

```diff
@@ -0,0 +1,481 @@
+              to be done which may increase the time to generate the inventory.
+        type: bool
+        default: False
+    fail_on_duplicate:
+        description:
+            - If True, the vpc_inventory plugin will fail if it finds duplicate host names or IP addresses. This can
+              occur when targetting different VPCs and regions.
+        type: bool
+        default: True
+    ansible_display_name:
+        description: By default, the VSI's name will be used as the name displayed by Ansible in the output.
+          Instead of the VSI's name, its IP address or ID may also be used.
+        type: str
+        choices: [name, ip, id]
+        default: "name"
+    ansible_host_type:
+        description: Determines if the IP Address or the VSI name will be used as the "ansible_host" variable
+          in playbooks.
+        type: str
+        choices: [name, ip]
+        default: "ip"
+'''
+
+EXAMPLES = '''
+# The most minimal example, targeting only a single region
+plugin: ibm.cloudcollection.vpc_inventory
+regions:
+  - us-south
+
+# Target multiple regions and only add running VSIs to the inventory
+plugin: ibm.cloudcollection.vpc_inventory
+regions:
+  - us-south
+  - br-sao
+filters:
+  status: 'running'
+
+# Generate an inventory of all running VSIs, using floating IP addresses and adding extra host variables.
+plugin: ibm.cloudcollection.vpc_inventory
+regions:
+  - us-south
+  - br-sao
+filters:
+  status: 'running'
+compose:
+  memory: memory
+  image: image
+  profile: profile
+  vpc: vpc
+use_floating_ips: True
+
+# Generate an inventory of running VSIs that were tagged with both 'database' and 'application.'
+# Additionally, create a db2 group for any VSI that has 'db2' as part of its name.
+plugin: ibm.cloudcollection.vpc_inventory
+regions:
+  - us-south
+  - br-sao
+filters:
+  status: 'running'
+  tags:
+    - database
+    - application
+groups:
+  db2: "'db2' in name"
+compose:
+  vpc: vpc
+  tags: tags
+
+# Generate an inventory of running systems that excludes VSI by IP, name, and ID.
+plugin: ibm.cloudcollection.vpc_inventory
+regions:
+  - us-south
+filters:
+  status: 'running'
+exclude_ip:
+  - 10.x.x.x
+  - 11.x.x.x
+exclude_name:
+  - example-name-1
+  - example-name-2
+exclude_id:
+  - abc_example_id_xyz
+
+# Generate an inventory of running VSIs, grouped by their memory value. The groups will be prefixed with 'custom.'
+plugin: ibm.cloudcollection.vpc_inventory
+regions:
+  - us-south
+  - br-sao
+filters:
+  status: 'running'
+keyed_groups:
+  - prefix: custom
+    key: memory
+compose:
+  memory: memory
+'''
+
+import xml.etree.ElementTree as ET
+import json
+import sys
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
+from ansible.module_utils.six import string_types, viewitems, reraise
+from ansible.errors import AnsibleParserError
+from ansible_collections.ibm.cloudcollection.plugins.module_utils.ibmcloud import Terraform, ibmcloud_terraform
```

Comment:
```suggestion
from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable
from ansible.module_utils.six import string_types, viewitems
from ansible.errors import AnsibleParserError
from ansible_collections.ibm.cloudcollection.plugins.module_utils.ibmcloud import ibmcloud_terraform
```

Some of these imports are unused.

Reply:
Thank you @jaywcarman for the review and the suggested changes. I'm removed the unused imports.